onnx/defs/tensor/old.cc

```diff
@@ -3192,7 +3192,14 @@ ONNX_OPERATOR_SET_SCHEMA(
               starts.size() != ends.size()) {
             fail_shape_inference(
                 "Incorrect or missing attribute value for starts and ends");
-            ;
+          }
+          auto is_negative = [](int64_t index) {
+            return index < 0;
+          };
+          if (!std::none_of(starts.begin(), starts.end(), is_negative)) {
+            fail_shape_inference("Attribute start cannot have negative value.");
+          } else if (!std::none_of(ends.begin(), ends.end(), is_negative)) {
+            fail_shape_inference("Attribute end cannot have negative value.");
```

Comment:
Instead of throwing an error can we either work with the negative value (when possible) or fallbacl to rank inference? This is because some runtimes (example ORT) did allow negative values even in cases where the standard did not explicitly state it... I am afraid throwing an error here will break existing scenarios...

Reply:
Good point. After offline discussion, to prevent breaking existing old models, ONNX should at least perform rank inference instead of throwing an error. Just updated. Also it can resolve the shape inference error from #2281. Please review it again. Thank you!

Reply:
You said: Also it can resolve the shape inference error from #2281. Please review it again.
Which error?

Reply:
Is this change complete? I only see you switched from throwing error for attr axis and not for start and end

Reply:
Oh sorry. I meant this one: https://github.com/onnx/onnx/issues/3565 Also update start and end. Thank you!